services/orders.js

```diff
@@ -1,8 +1,96 @@
 const moment = require('moment');
 const { Op } = require('sequelize');
 const { Order, OrderItem } = require('../models');
+const { take, countBy, meanBy, chain, sortBy } = require('lodash');
 
-function storeOrder({ grantAmount, orderReason, postcodeArea, items }) {
+function getAllOrders() {
+    return Order.findAll({
+        order: [['updatedAt', 'DESC']],
+        include: [
+            {
+                model: OrderItem,
+                as: 'items'
+            }
+        ]
+    }).then(orders => {
+        const normalisedDateFormat = 'YYYY-MM-DD';
+        // the date we added dropdown grant size options rather than free text
+        const LAUNCH_DATE = '2018-02-01';
+
+        let items = orders.reduce((acc, cur) => acc.concat(cur.items), []);
+
+        let totalOrders = orders.length;
+        let averageProductsPerOrder = Math.round(meanBy(orders, 'items.length'));
+        let averageItemQuantityPerOrder = Math.round(meanBy(orders, o => o.items.reduce((acc, cur) => acc + cur.quantity, 0)));
+
+        let orderByCount = arr => {
+            return chain(arr).map((count, code) => {
+                return {
+                    code: code,
+                    count: count
+                };
+            }).sortBy('count').reverse().value();
+        };
+
+        // sum items by their quantity ordered
+        let itemsByQuantity = items.reduce((acc, cur) => {
+            if (!acc[cur.code]) {
+                acc[cur.code] = 0;
+            }
+            acc[cur.code] += cur.quantity;
+            return acc;
+        }, {});
+
+        // order items by total ordered
+        let mostPopularItemsByQuantity = orderByCount(itemsByQuantity);
+        let mostPopularItems = take(orderByCount(countBy(items, 'code')), 10);
+
+        let orderReasons = orderByCount(countBy(orders, 'orderReason')).filter(o => {
```

Comment:
This could possibly be flattened up a bit using `lodash/fp`, something like:

```js
const { filter } = require('lodash/fp');

const filterIsRecentOrder = filter(_ => moment(_.createdAt).isAfter(LAUNCH_DATE));
const filterHasCode = filter(_ => _.code !== '');

const orderReasons = filterHasCode(orderByCount(countBy(orders, 'orderReason')));

const ordersAfterLaunch = filterIsRecentOrder(orders);
const grantAmounts = filterHasCode(orderByCount(countBy(ordersAfterLaunch, 'grantAmount')));
```